src/parser/hunter/beastmastery/modules/items/QaplaEredunWarOrder.tsx

```diff
@@ -0,0 +1,108 @@
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import React from 'react';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import SPELLS from 'common/SPELLS';
+import Events from 'parser/core/Events';
+import { QAPLA_KILL_COMMAND_REDUCTION_MS } from 'parser/hunter/beastmastery/constants';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import { formatNumber, formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+
+/**
+ * Barbed Shot reduces the cooldown of Kill Command by 4.0 sec.
+ *
+ * Example log:
+ *
+ */
+class QaplaEredunWarOrder extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  effectiveKCReduction: number = 0;
+  wastedKCReduction: number = 0;
+  wastedCasts: number = 0;
+
+  protected spellUsable!: SpellUsable;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.QAPLA_EREDUN_WAR_ORDER_EFFECT.bonusID);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.BARBED_SHOT), this.onBarbedShotCast);
+  }
+
+  get totalPossibleCDR() {
```

Comment:
Extremely unlikely but you should probably put a check to make sure you don't divide by zero (i know its like almost impossible as a bm to not kill command but its a good idea to avoid it even if edge case)
(i say it here since all division happens off of this)